### DISCUSSION TITLE
Add a new partition for flashing emmc

### DISCUSSION THREAD

**Discussion by elbbit01** [OE4T/meta-tegra#1251] [General]
Hello, 
for context, I am using kirkstone with l4t 35.2.1.

I have added an extra partition (backup.ext4.img) at my flash_t194_sdmmc_rootfs_ab.xml and all was working fine. We decided that we want it new flashed every time we "do_flash.sh" so I added also a file_name for it.

I followed the steps from this older discussion https://github.com/OE4T/meta-tegra/discussions/858#discussioncomment-1664607 but without the line `TEGRA_BUPGEN_STRIP_IMG_NAMES = "app.ext4.img"` since I want it directly in the flash tarball and not in any update payload.

At the rootfs_ab.xml file:
```

+	<partition name="BackupPartitionEnc" type="data">
+            <allocation_policy> sequential </allocation_policy>
+            <filesystem_type> basic </filesystem_type>
+            <size> 10737418240 </size>
+            <file_system_attribute> 0 </file_system_attribute>
+            <allocation_attribute> 0x8 </allocation_attribute>
+            <align_boundary> 4096 </align_boundary>
+	    <percent_reserved> 0 </percent_reserved>
+            <filename> backup.ext4.img </filename>    
+            <description> **Required.** Contains the enrypted backup partitions for System Data A, System Data B and K3S Config.
+              This partition must be defined after `primary_gpt` so it can be accessed as the fixed known special device
+              `/dev/mmcblk0p4`. </description>
+        </partition>	

```

At  my recept under images:
```

SUMMARY = "Backup image for flashing"
HOMEPAGE = ""
LICENSE = "CLOSED"

S = "${WORKDIR}"

do_install(){
    mkdir -p ${STAGING_DIR}/backup-image
    dd if=/dev/zero of=backup.ext4.img bs=1 count=0 seek=10737418240
    /sbin/mkfs.ext4 backup.ext4.img
    install -m 0644 backup.ext4.img ${STAGING_DIR}/backup-image/backup.ext4.img
}

PACKAGES = "${PN}"
```

And at my image file:

```
# Include the image for the backup partition in emmc
TEGRAFLASH_PKG_DEPENDS += "backup-partition-image:do_populate_sysroot"

tegraflash_custom_pre () {
    cp ${STAGING_DIR}/backup-image/backup.ext4.img backup.ext4.img
}

TEGRA_BUPGEN_STRIP_IMG_NAMES = "backup.ext4.img"
```
----------------------------------------------------------
**I get as an error, that tegraparser_v2 gets an erro: file backup.ext4.img open failed**

Am I missing some alternative variable for the tegraflash? Has anybody of you a hint?

Thanks in advance,
Alvaro.

**Comment by madisongh** [comment]
> I get as an error, that tegraparser_v2 gets an erro: file backup.ext4.img open failed

If it's during the build, what recipe was building when the error was generated?  The `TEGRA_BUPGEN_STRIP_IMG_NAMES` setting has to be visible in that recipe.  You could do that by moving that setting to your machine config file (if you've defined a custom machine), or through a `bbappend` for the recipe in question (which I suspect is `tegra-minimal-initramfs`).

If it's happening during flashing, then double-check the modifications to your image recipe.  They look right to me, though.

**Reply by elbbit01** [reply]
Ah yes, excuse me. 

It is in build time and it is as you suspect on tegra-minimal-initramfs.bb:do_image_cpio for more details. 

Of course, I had that variable setting on my image configuration, I moved it to the machine config and it worked.

Thanks a lot!

**Comment by Islam-Hussein-11** [comment]
Hi I'm facing similar error based also on the same branch. 


| [   3.3129 ] tegraparser_v2 --pt flash.xml.bin --generateflashindex /home/isalm/Desktop/mender-tegra/tegra-demo-distro/build/tmp/work/jetson_xavier_nx_devkit_emmc-oe4t-linux/tegra-minimal-initramfs/1.0-r0/bup-payload/signed/flash.xml.tmp flash.idx
| [   3.3151 ] File SMDFILE open failed
| Error: Return value 19
| Command tegraparser_v2 --pt flash.xml.bin --generateflashindex /home/isalm/Desktop/mender-tegra/tegra-demo-distro/build/tmp/work/jetson_xavier_nx_devkit_emmc-oe4t-linux/tegra-minimal-initramfs/1.0-r0/bup-payload/signed/flash.xml.tmp flash.idx
| WARNING: exit code 1 from a shell command.
ERROR: Task (/home/isalm/Desktop/mender-tegra/tegra-demo-distro/layers/meta-tegra/recipes-core/images/tegra-minimal-initramfs.bb:do_image_cpio) failed with exit code '1'
NOTE: Tasks Summary: Attempted 4927 tasks of which 4926 didn't need to be rerun and 1 failed.
Summary: 1 task failed:
  /home/isalm/Desktop/mender-tegra/tegra-demo-distro/layers/meta-tegra/recipes-core/images/tegra-minimal-initramfs.bb:do_image_cpio
Summary: There were 8 WARNING messages.
Summary: There was 1 ERROR message, returning a non-zero exit code.


Any help would be appreciated.

This error happens when I build I'm trying to build tegrademo-mender but based on Jetpack 5 which uses l4t-35.x.x. I could build already the one which was based on l4t-32.7.x. I've used also the PR-19 which is still open. I'm just trying to generate a minimal Image which is based on kirkstone branch to migrate it with the running project.

**Reply by dwalkes** [reply]
> I've used also the PR-19 which is still open.

Probably should move the discussion to PR https://github.com/OE4T/meta-mender-community/pull/19

Some notes there about what you tried and how this compares to what others have successfully used on the branch would be good.  I'm guessing the answer to this particular problem is related to the layout in the flash.xml, and potentially an issue with the size of partitions relative to the size of disk.
